modules/reactor/doc/content/source/meshgenerators/PinMeshGenerator.md

```diff
@@ -0,0 +1,47 @@
+The `PinMeshGenerator` object adopts much of the existing input structure of `PolygonConcentricCircleMeshGenerator`](PolygonConcentricCircleMeshGenerator.md) but uses parameters that are more typical for reactor design.
+
+## Block ID Information
+The [!param](/Mesh/PinMeshGenerator/region_ids) parameter is used to identify regions within the pin that belong to the same block both radially and axially. This functionality is intended for easy identification of regions within the mesh that will have the same properties, such as material assignments, and will assign the region ID to both the subdomain (block) ID and name.
+This parameter can be directly input, or can be automated with the [!param](/Mesh/ReactorMeshParams/procedural_region_ids) in the [`ReactorMeshParams`](ReactorMeshParams.md) MeshGenerator input for [!param](/Mesh/PinMeshGenerator/reactor_params). If explicit assignment is being used it should be noted that if [!param](/Mesh/PinMeshGenerator/quad_center_elements) is false than the center most radial mesh division needs to be its own region due to element type restrictions for block IDs.
+In the case of user-defined subdomain ID assignment, [!param](/Mesh/PinMeshGenerator/region_ids) should be given as a vector starting from the center-most region, or in the case of a three dimensional problem, a two dimensional array with each component vector being the IDs for the appropriate axial region starting from the bottom of the geometry. 
+The automated ID assignment, via the [!param](/Mesh/ReactorMeshParams/procedural_region_ids) option, concatenates the [!param](/Mesh/PinMeshGenerator/pin_type) and the radial mesh region number (indexed starting at 0 for the innermost region). If radial regions are used, the innermost meshed interval will always be made a unique region to allow for either quad or tri center elements. In the case that the concatenation exceeds the type limit, the lowest digits of the [!param](/Mesh/PinMeshGenerator/pin_type) will be truncated. The same ID will be used for all axial layers in three dimensional meshes.
+
+For example a pin with a type ID of 5 with two rings a duct would procedurally generate four region IDs:
+- The center-most ring (assuming it only has a single mesh interval) region ID of 50
+- The second ring region ID of 51
+- The background region ID of 52
+- The duct region ID of 53
+
+If the same pin instead has a type ID of 65535 the same four regions would be assigned as:
+- The center-most ring region ID of 65530
+- The second ring region ID of 65531
+- The background region ID of 65532
+- The duct region ID of 65533
+
+
+## Reporting ID Information
+The `PinMeshGenerator` object also tags the mesh elements with the reporting ID named "region_id".
+The `PinMeshGenerator` object also automatically tags the mesh with the [!param](/Mesh/PinMeshGenerator/pin_type) using the name "pin_type_id" and, if extruded, the axial layers using the name "plane_id".
+
+## Exterior Boundary ID Information
```

Comment:
this will conflict with the add 0s strategy, since it's 20000, so 2 +  max number of 0s + the duct id